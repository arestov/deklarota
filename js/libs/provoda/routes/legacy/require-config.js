import requirejs from 'requirejs'
import path from 'path'

requirejs.config({
  baseUrl: __dirname,
  map: {
		'*': {
			spv: path.join(process.cwd(), 'js/libs/spv.js'),
      pv: path.join(process.cwd(), 'js/libs/provoda/provoda'),
			__lib: path.join(process.cwd(), 'js/libs/provoda'),
		}
	},
})

module.exports = requirejs
