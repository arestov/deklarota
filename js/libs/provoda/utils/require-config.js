var requirejs = require('requirejs')
var path = require('path')

requirejs.config({
  baseUrl: __dirname,
  map: {
		'*': {
			spv: path.join(process.cwd(), 'js/libs/spv.js'),
      pv: path.join(process.cwd(), 'js/libs/provoda/provoda'),
			__lib: path.join(process.cwd(), 'js/libs/provoda'),
		}
	},
})

module.exports = requirejs
