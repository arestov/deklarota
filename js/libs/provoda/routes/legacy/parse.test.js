
var test = require('ava')
var requirejs = require('requirejs')
require('./require-config')
var parse = requirejs('./parse')

test('check parsing', function(t) {
  'use strict'

  t.snapshot(parse('tracks/[:artist],[:track]'))
})
