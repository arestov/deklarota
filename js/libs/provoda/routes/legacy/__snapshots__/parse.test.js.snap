// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse should parse "tracks/[:artist],[:track]" correctly 1`] = `
Object {
  "clean_string_parts": Array [
    "tracks/",
    ",",
    "",
  ],
  "states": Array [
    "artist",
    "track",
  ],
  "states_map": Array [
    Array [
      "artist",
      "artist",
      undefined,
    ],
    Array [
      "track",
      "track",
      undefined,
    ],
  ],
}
`;
